packages/Popover/src/hooks.ts

```diff
@@ -56,3 +56,18 @@ export function useMutationObserver<Value>(
 
   return value;
 }
+
+export function useElementNode<ElementType = HTMLElement>(): [
+  ElementType | null,
+  (node: ElementType | null) => void
+] {
+  const [element, setElement] = useState<ElementType | null>(null);
+
+  const elementRefCallback = useCallback(node => {
+    if (node !== null) {
+      setElement(node as ElementType);
+    }
+  }, []);
+
+  return [element, elementRefCallback];
```

Comment:
It would really be sweet if we could offer an API similar to `useRef`... Haven't thought through how that would look from the implementation side without actually using `useRef` in here though.